backend/handlers/get_rule_graph.go

```diff
@@ -0,0 +1,69 @@
+package handlers
+
+import (
+	"github.com/Zeus-Labs/ZeusCloud/rules/types"
+	"log"
+	"net/http"
+
+	"github.com/Zeus-Labs/ZeusCloud/rules/attackpath"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j"
+)
+
+var ruleGraphMap = map[string]types.Rule{
+	"private_serverless_admin_permissions":          attackpath.PrivateServerlessAdmin{},
+	"private_vm_admin_permissions":                  attackpath.PrivateVmAdmin{},
+	"publicly_exposed_vm_admin_permissions":         attackpath.PubliclyExposedVmAdmin{},
+	"publicly_exposed_vm_high_permissions":          attackpath.PubliclyExposedVmHigh{},
+	"publicly_exposed_vm_priv_escalation":           attackpath.PubliclyExposedVmPrivEsc{},
+	"publicly_exposed_serverless_admin_permissions": attackpath.PubliclyExposedServerlessAdmin{},
+	"publicly_exposed_serverless_high_permissions":  attackpath.PubliclyExposedServerlessHigh{},
+	"publicly_exposed_serverless_priv_escalation":   attackpath.PubliclyExposedServerlessPrivEsc{},
+	"third_party_admin_permissions":                 attackpath.ThirdPartyAdmin{},
+	"third_party_high_permissions":                  attackpath.ThirdPartyHigh{},
+}
+
+func GetRuleGraph(driver neo4j.Driver) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("Entered Get Rule Graph \n")
+		resourceId := r.URL.Query().Get("resource")
+		rule := r.URL.Query().Get("rule")
+
+		if _, ok := ruleGraphMap[rule]; !ok {
+			log.Println("Invalid rule to graph provided")
+			http.Error(w, "Invalid rule to graph provided", 500)
+			return
+		}
+
+		session := driver.NewSession(neo4j.SessionConfig{
+			AccessMode:   neo4j.AccessModeRead,
+			DatabaseName: "neo4j",
+		})
+		defer session.Close()
+
+		// TODO: parse the return
+		_, err := session.ReadTransaction(func(tx neo4j.Transaction) (interface{}, error) {
+			ruleToCall := ruleGraphMap[rule]
+			graphPathResult, err := ruleToCall.ProduceRuleGraph(tx, resourceId)
+			if err != nil {
+				log.Printf("failed to retrieve rule graph results")
+				http.Error(w, "failed to retrieve rule graph results", 500)
+				return graphPathResult, err
+			}
+			ruleToCall.ProduceDisplayGraph(graphPathResult)
```

Comment:
ProduceDisplayGraph should not be rule specific. Given list of paths it's creating adjacency list. And for prettier display purposes it's getting rid of some nodes